hubble/pkg/printer/terminal.go

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Hubble
+
+package printer
+
+import (
+	"fmt"
+	"io"
+	"strings"
+)
+
+type terminalEscaperBuilder struct {
+	replacer *strings.Replacer
+}
+
+// newTerminalEscaperBuilder creates a new terminalEscaperBuilder that allows a subset of control
+// sequences, such as a reserved set of colors and their reset sequences.
+func newTerminalEscaperBuilder(allowed []string) *terminalEscaperBuilder {
+	var oldnew []string
+	for _, a := range allowed {
+		oldnew = append(oldnew, a, a)
+	}
+	oldnew = append(oldnew, "\x1b", "^[", "\r", "\\r")
+	return &terminalEscaperBuilder{replacer: strings.NewReplacer(oldnew...)}
+}
+
+func (teb *terminalEscaperBuilder) NewWriter(w io.Writer) *terminalEscaperWriter {
+	return &terminalEscaperWriter{w: w, replacer: teb.replacer}
+}
+
+// terminalEscaperWriter replaces ANSI escape sequences and other terminal special
+// characters to avoid terminal escape character attacks. It stops on the first error
+// encountered and stores its value. The caller is responsible for checking Err()
+// when done writing.
+type terminalEscaperWriter struct {
+	w        io.Writer
+	replacer *strings.Replacer
+	err      error
+}
+
+func (tew *terminalEscaperWriter) Write(a ...interface{}) {
```

Comment:
nit: naming a method `Write` when it doesn't implement `io.Writer` might be confusing. Also it probably doesn't need to be exported since it's not imported outside this package.

Reply:
Yeah its the names that were there before. I made it public to differentiate the available API, but I also wanted to rename those to Print and Printf since they map to fmt and not writer.

I'll update

Reply:
Updated to use the print funcs format